cellxgene_resources/curation_sample_code.ipynb

```diff
@@ -747,50 +747,54 @@
-    "    pattern3 = '(gene_name)'\n",
-    "    pattern4 = 'gene_name'\n",
-    "    pattern5 = '([\\w\\d\\-]+)'\n",
-    "    split1 = row_8.str.split(pattern1, expand=True).dropna(axis=1)\n",
-    "    gene_id_col = split1.loc[:,split1.iloc[0].str.contains(pattern2)].columns\n",
-    "    gene_sym_col = split1.loc[:,split1.iloc[0].str.contains(pattern3)].columns\n",
-    "    gene_ids = split1[gene_id_col].iloc[:,0].str.split(pattern2, expand=True)[1]\n",
-    "    gene_sym = split1[gene_sym_col].iloc[:,0].str.split(pattern4, expand=True)[1].str.split(pattern5, expand=True)[1]\n",
-    "    gtf_df = pd.DataFrame({'gene_ids':gene_ids, 'gene_symbols':gene_sym})\n",
-    "    gtf_df.dropna(axis=0, inplace=True)\n",
-    "    gtf_df.set_index('gene_ids')\n",
-    "    return gtf_df"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "# Filters out gene_symbols with multiple gene_ids and assigns 'multiple' as gene_id to corresponding gene_symbol\n",
-    "def assign_multiple(gtf_df):\n",
-    "    '''\n",
-    "    Input: gtf pandas dataframe with gene_id and gene_symbol columns\n",
-    "    Output: de-duplicated dataframe\n",
-    "    '''\n",
-    "    dups = gtf_df[gtf_df.duplicated(subset='gene_symbols',keep=False)]\n",
-    "    dups['gene_ids'] = 'multiple'\n",
-    "    dups = dups.drop_duplicates()  # only keep one instance of the gene_symbol\n",
-    "    gene_df_wo_dups = gtf_df.drop_duplicates(subset='gene_symbols',keep=False) \n",
-    "    gene_df_multiple = gene_df_wo_dups.append(dups)  # create new df with all non-duplicated gene_symbols and gene_symbols with multiple ensembl ids\n",
-    "    gene_df_multiple.set_index('gene_ids', inplace=True)\n",
-    "    return gene_df_multiple"
+    "   \n",
+    "    features = df[2].unique()\n",
+    "    print(features)\n",
+    "    feature = str(input('Which feature should be extracted?'))\n",
+    "    ens_id_prefixes = ['ENSMUSG', 'ENSG']\n",
+    "    ens_id_list = []\n",
+    "\n",
+    "    try:\n",
+    "        if feature in features:\n",
+    "            print(f'Pulling out {feature}...')\n",
+    "            feature_info = df[df[2] == feature][8]\n",
+    "            df1 = feature_info.str.split(';', expand = True).dropna(axis = 1)\n",
+    "            print('Need to id gene_symbols and ensg_id cols, here\\'s the first row:\\n')\n",
+    "            for k,v in df1.head(1).to_dict('records')[0].items():\n",
+    "                print(str(k),':',v)  # caution: relevant columns with gene symbols and ensg ids may change for the specific gtf\n",
+    "            gene_id_col = int(input('Column number for ensg_ids?\\n'))\n",
+    "            gene_symbol_col = int(input('Column number for gene_symbols?\\n'))\n",
+    "\n",
+    "            for p in ens_id_prefixes:\n",
+    "                if p in df1[gene_id_col][0]:\n",
+    "                    print(f'Extracting {p} ids...')\n",
```

Comment:
I think you could remove this print statement.
Right now, you enter a few options and then you see this that ends with `...` so you kinda wait for something after that.